### ISSUE TITLE
MTProxy not working anymore

### ISSUE THREAD

**Issue by davidbaumann** [TelegramMessenger/MTProxy#197]
Hello,

actually after 2 weeks running on a docker container on my german vps, mtproxy stopped working (client says "connecting" forever).
I double checked the secret. I compiled from github today.

I even installed in a local virtual machine to check closer, with a new secret and a fresh pull from github.
According to wireshark, it connects to the telegram proxies from my virtual machine, and my telegram desktop client on windows connects to the local mtproxy instance (connecting).

Long story short: I tried different ways, and now I don't have any idea how to solve my problem.
curl localhost:8888 gives me a 404. 
Also I double checked port 443 really points to my mtproxy. When I kill mtproxy, it says "reconnecting in... seconds", when mtproxy is running, it shows "connecting" forever.

I have no idea what to try.

When I direct connect without mtproxy, telegram just runs fine. I am out of options and ideas now :(

Thanks.

**Comment by epm157** [comment]
Hi David,
Would you please install htop and attach an image showing the Mtproto Proxy process running?

**Comment by mtimvik** [comment]
I have the same issue

**Comment by epm157** [comment]
Is it an option to talk to you guys in skype or remote desktop to help to sort of the issue? Because writing back and forth here does not provide enough information.

**Comment by Am6er** [comment]
Hi, i've got the same behaviour. 
Proxy was up and running from 6-th of July. Yesterday i've got "connecting..." problem. I tried to get new secret, change port - the same "connecting...".

Image htop
![](https://image.ibb.co/cxPyhA/2018-11-22-17-12-56.png)

**Comment by kooran** [comment]
test it with a vpn on client device , if it works, your server IP has been blocked

**Comment by Am6er** [comment]
Tested it with vpn on client device as well. "Connecting...". Try to telnet on server list from proxy-multi.conf - telnet succeed.

**Comment by Am6er** [comment]
I run mtproxy with -v flag for verbose output, and in log file i saw errors due connection to upstream Telegram servers:

$ cat nohup.out | grep -v "New connection" | grep -v "Creating NEW" | grep -v "Disconnected from RPC Middle-End" | grep -v "epoll_ready=" | grep -v "Closing connection socket" | less

[11912][2018-11-27 13:11:24.595723 local] Invoking engine mtproxy-0.01 compiled at Nov 27 2018 12:36:15 by gcc 4.4.7 201
20313 (Red Hat 4.4.7-23) 32-bit after commit f9158e3129efd4ccdc291aefb840209791226a77
[11912][2018-11-27 13:11:24.596154 local] config_filename = 'proxy-multi.conf'
[11912][2018-11-27 13:11:24.596221 local] /etc/hosts changed, reloaded
[11912][2018-11-27 13:11:24.596358 local] config loaded!
[11912][2018-11-27 13:11:24.596533 local] loaded 128 bytes from password file proxy-secret
[11912][2018-11-27 13:11:24.596906 local] using main IP __.__.__.__/24 at interface eth0
[11912][2018-11-27 13:11:24.597024 local] using main IP 0:::::::/0 at interface (null)
[11912][2018-11-27 13:11:24.597674 local] Started as [__.__.__.__:8888:11912:1543313484]
[11912][2018-11-27 13:11:24.597970 local] configuration file proxy-multi.conf re-read successfully (809 bytes parsed), new configuration active
[11912][2018-11-27 13:11:24.598048 local] opened tcp socket
[11912][2018-11-27 13:11:24.598069 local] Server started
[11912][2018-11-27 13:11:24.598116 local] main loop
[11912][2018-11-27 13:11:25.429841 local] trying to determine connection type
[11912][2018-11-27 13:11:25.429969 local] invalid "random" 64-byte header, entering global skip mode
[11912][2018-11-27 13:11:26.808263 local] got error for tcp socket #35, [149.154.165.250]:8888 : Connection refused
[11912][2018-11-27 13:11:26.938909 local] trying to determine connection type
[11912][2018-11-27 13:11:26.938947 local] tcp opportunistic encryption mode detected, tag = efefefef, target=4
[11912][2018-11-27 13:11:26.938980 local] ext_rpcs_execute: cannot forward mtproto packet
[11912][2018-11-27 13:11:27.015449 local] got error for tcp socket #15, [149.154.165.250]:8888 : Connection refused
[11912][2018-11-27 13:11:27.120083 local] trying to determine connection type
[11912][2018-11-27 13:11:27.120117 local] invalid "random" 64-byte header, entering global skip mode
[11912][2018-11-27 13:11:27.270968 local] got error for tcp socket #11, [149.154.165.250]:8888 : Connection refused
[11912][2018-11-27 13:11:27.367479 local] trying to determine connection type
[11912][2018-11-27 13:11:27.367526 local] invalid "random" 64-byte header, entering global skip mode
[11912][2018-11-27 13:11:27.435896 local] got error for tcp socket #60, [149.154.165.250]:8888 : Connection refused
[11912][2018-11-27 13:11:27.928693 local] got error for tcp socket #20, [149.154.165.250]:8888 : Connection refused
[11912][2018-11-27 13:11:28.208462 local] got error for tcp socket #12, [149.154.165.250]:8888 : Connection refused
[11912][2018-11-27 13:11:28.325062 local] got error for tcp socket #50, [149.154.165.250]:8888 : Connection refused
[11912][2018-11-27 13:11:28.391433 local] got error for tcp socket #26, [149.154.165.250]:8888 : Connection refused
[11912][2018-11-27 13:11:28.397687 local] trying to determine connection type
.....

and so on.

**Comment by Am6er** [comment]
Installed on the SAME host socks5 proxy - Telegram works fine. So there is no blocking issue.

**Comment by edward-p** [comment]
Check the time of your system, make sure there's no inaccuracy. Always using ntp to sync your time with the Internet.

**Comment by Am6er** [comment]
Edward, thanks! Problem solved!

**Comment by summershrimp** [comment]
both time on server and phone are the same, but shows `invalid "random" 64-byte header, entering global skip mode`, and the client stays checking...
